core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -41,38 +43,37 @@ public CommitHandler(
   }
 
   public void commit(Snapshot snapshot) throws CommitException, UnknownTransactionStatusException {
-    prepare(snapshot, true);
+    try {
+      prepare(snapshot);
+    } catch (PreparationException e) {
+      abort(snapshot.getId());
+      rollbackRecords(snapshot);
+      if (e instanceof PreparationConflictException) {
+        throw new CommitConflictException(e.getMessage(), e);
+      }
+      throw new CommitException(e.getMessage(), e);
+    }
+
     preCommitValidation(snapshot, true);
     commitState(snapshot);
     commitRecords(snapshot);
   }
 
-  public void prepare(Snapshot snapshot, boolean abortIfError)
-      throws CommitException, UnknownTransactionStatusException {
-    String id = snapshot.getId();
+  public void prepare(Snapshot snapshot) throws PreparationException {
```

Comment:
Made this `prepare()` method throw `PreparationException` instead of `CommitException`.